src/meta-srv/src/procedure/wal_prune.rs

```diff
@@ -131,15 +201,66 @@ impl WalPruneProcedure {
             return Ok(Status::done());
         }
 
-        // Safety: `flush_entry_ids_map` are not empty.
-        self.data.min_flushed_entry_id = *(flush_entry_ids_map.values().min().unwrap());
+        let min_max_result = flush_entry_ids_map.values().minmax();
+        let max_flushed_entry_id = match min_max_result {
+            MinMaxResult::NoElements => {
+                return Ok(Status::done());
+            }
+            MinMaxResult::OneElement(flushed_entry_id) => {
+                self.data.min_flushed_entry_id = *flushed_entry_id;
+                *flushed_entry_id
+            }
+            MinMaxResult::MinMax(min_flushed_entry_id, max_flushed_entry_id) => {
+                self.data.min_flushed_entry_id = *min_flushed_entry_id;
+                *max_flushed_entry_id
+            }
+        };
+        if let Some(threshold) = self.data.trigger_flush_threshold {
+            for (region_id, flushed_entry_id) in flush_entry_ids_map {
+                if flushed_entry_id + threshold < max_flushed_entry_id {
+                    self.data.regions_to_flush.push(region_id);
+                }
+            }
+            self.data.state = WalPruneState::FlushRegion;
+        } else {
+            self.data.state = WalPruneState::Prune;
+        }
+        Ok(Status::executing(true))
+    }
+
+    /// Send the flush request to regions with low flush entry id.
+    pub async fn on_sending_flush_request(&mut self) -> Result<Status> {
+        let peer_to_region_ids_map = self
+            .build_peer_to_region_ids_map(&self.context, &self.data.regions_to_flush)
+            .await?;
+        let flush_instructions = self.build_flush_region_instruction(peer_to_region_ids_map)?;
+        for (peer, flush_instruction) in flush_instructions.into_iter() {
+            let msg = MailboxMessage::json_message(
+                &format!("Flush regions: {}", flush_instruction),
+                &format!("Metasrv@{}", self.context.server_addr),
+                &format!("Datanode-{}@{}", peer.id, peer.addr),
+                common_time::util::current_time_millis(),
+                &flush_instruction,
+            )
+            .with_context(|_| error::SerializeToJsonSnafu {
+                input: flush_instruction.to_string(),
+            })?;
+            let ch = Channel::Datanode(peer.id);
+            // Instantly drop the receiver to not block.
+            let recv = self.context.mailbox.send(&ch, msg, FLUSH_TIMEOUT).await?;
```

Comment:
For no reply instruction, I think we can set the timeout to 0s. cc @fengjiachun

Reply:
It is a one-way RPC; the mailbox needs to handle this. I suggest adding a `oneWay` method for the mailbox and eliminating the timeout logic.